### ISSUE TITLE
multi-stage training

### ISSUE THREAD

**Issue by baibaidj** [open-mmlab/mmsegmentation#56]
**Describe the feature**
When transferring pre-trained encoders to a segmentation setting, e.g. Unet, one may want to first freeze the encoder and warmup the newly-added decoder  for several epochs (stage 1); then unfreeze the encoder and train the whole network together (stage 2). 

**Motivation**
Ex1. This multi-stage training scheme supposedly results in better performance. 
Ex2. It can also facilitate easy transition from FP32 training to FP16 because sometimes using FP16 from right beginning may fail due to large loss that cannot be represented by FP16. 

**Related resources**

**Additional context**

**Comment by xvjiarui** [comment]
Hi @baibaidj 
The `encoder` could be frozen by `frozen_stages` arguments in the backbone. 
In the current implementation, you may have to manually set `frozen_stages=3` and train a few epochs (stage 1). Then set `frozen_stages=-1` and train the rest epochs (stage 2). 
We will consider this feature and we don't have a plan for it yet.

**Comment by baibaidj** [comment]
> Hi @baibaidj
> The `encoder` could be frozen by `frozen_stages` arguments in the backbone.
> In the current implementation, you may have to manually set `frozen_stages=3` and train a few epochs (stage 1). Then set `frozen_stages=-1` and train the rest epochs (stage 2).
> We will consider this feature and we don't have a plan for it yet.

@xvjiarui Thanks.
